server/handlers.go

```diff
@@ -467,7 +467,7 @@ func (s *Server) handleConnectorCallback(w http.ResponseWriter, r *http.Request)
 
 	if err != nil {
 		s.logger.Errorf("Failed to authenticate: %v", err)
-		s.renderError(r, w, http.StatusInternalServerError, fmt.Sprintf("Failed to authenticate: %v", err))
+		s.renderError(r, w, http.StatusUnauthorized, fmt.Sprintf("Failed to authenticate: %v", err))
```

Comment:
This is not a good idea. This is where we catch unexpected errors as well. If there is a need to return 401 based on the error, the error needs to be checked for such conditions.

Reply:
Completely agree. I was not intending to add this to  the PR actually. So, are you suggesting we create a custom Error  type with code + error message and then check for `code` to decide whether to return `StatusInternalServerError` or `StatusUnauthorized` or are you saying that we check for the error message string and then decide whether to return `StatusInternalServerError` or `StatusUnauthorized`?

Reply:
Let's just revert this change. From the PR description, I don't think this is required.

Reply:
Done in 345eba905ddb2664855e79da2acf816ad6b97eb5